### PR TITLE
Unauthenticated Endpoints Session Id

### PR DESCRIPTION
**Pull request by xTEddie**

### CODE REVIEW

File: src/SDK.ts [hunk 1]
```diff
@@ -313,6 +318,12 @@ export default class SDK implements ISDK {
         const { data, headers } = response;
         this.setAuthCodeNonce(headers);
 
+        if (headers) {
+          if (headers[OmnichannelHTTPHeaders.ocSessionId.toLowerCase()]) {
```
Comment (elopezanaya) [inline src/SDK.ts:322]: is there a risk this could be null or undefined/. headers[OmnichannelHTTPHeaders.ocSessionId

File: src/SDK.ts [hunk 2]
```diff
@@ -374,6 +385,10 @@ export default class SDK implements ISDK {
     headers[OmnichannelHTTPHeaders.authenticatedUserToken] = authenticatedUserToken;
     headers[OmnichannelHTTPHeaders.authCodeNonce] = this.configuration.authCodeNonce;
 
+    if (this.sessionId) {
```
Comment (elopezanaya) [inline src/SDK.ts:388]: this block is repeated 12 times , any way we can avoid the repetition?

**Review by elopezanaya — CHANGES_REQUESTED** [review]

File: src/SDK.ts (hunk 2 above)
Comment (xTEddie) [reply src/SDK.ts]: Refactored

File: src/SDK.ts (hunk 1 above)
Comment (xTEddie) [reply src/SDK.ts:320]: It has a null/undefined check on whether the `headers` exists, then whether `Oc-Sessionid` header exists, then assign it accordingly.

**Review by elopezanaya — APPROVED** [review]

**Pull request merged**